2-With-broker/README.md

```diff
@@ -248,24 +262,55 @@ MSAL uses `canOpenURL:` to check if the broker is installed on the device. In
 
 #### Step 6: Register your RedirectUri in the application portal
 Using the broker adds an extra requirement on your redirectUri. The redirectUri _**must**_ have the following format:
+
+**For iOS:**
 ```CSharp
 $"msauth.{BundleId}://auth"
 ```
-**For example:**
+**Example for iOS:**
 ```CSharp
 public static string redirectUriOnIos = "msauth.com.yourcompany.XForms://auth"; 
 ```
-**You'll notice the RedirectUri matches the `CFBundleURLSchemes` name you included in the `Info.plist` file.**
+**You'll notice the iOS RedirectUri matches the `CFBundleURLSchemes` name you included in the `Info.plist` file.**
+
+**For Android:**
+```CSharp
+$"msauth://{Package Name}/{Signature Hash}"
+```
+**Example for Android:**
+```CSharp
+public static string redirectUriOnIAndroid = "msauth://UserDetailsClient.Droid/hgbUYHVBYUTvuvT&Y6tr554365466="; 
+```
+
+Once you have created your redirect URI, you can register it by navigating to the **Authentication** tab; under the **Redirect URIs** section, select **Public client/native (mobile & desktop)** entry from the dropdown box and paste in your redirect URI next to it. Click on save at the top of the page to save changes.
+
+**If you have the new experience**
+
+- **Android:** Select **Add a platform** under the **Platform Configurations** section, click on **Android** and add your package name and the signature hash.
 
-#### Step 7: make sure the redirect URI is registered with your app
+- **iOS:** Select **Add a platform** under the **Platform Configurations** section, click on **iOS** and add your iOS bundle id.
+
+Once you have the proper redirect URI, update the `BrokerRedirectUriOnIos` and the `BrokerRedirectUriOnAndroid` fields in the App.cs file with the values.
+
+#### Android Specific Redirect URI Configuration
+
+The redirect uri on Android will need to be created based on the signature of the .APK used to sign it. This means that it will be different depending on where this sample is run because Visual Studio creates a unique signing key for debugging purposes on every machine. You can figure out what that signature will be by running the following commands
+
+- For Windows: `keytool -exportcert -alias androiddebugkey -keystore %HOMEPATH%\.android\debug.keystore | openssl sha1 -binary | openssl base64`
```

Comment:
I can't get this to work, `openssl` is not installed by default on Windows.

Reply:
I don't think we should be recommending ppl to use `openssl` at all - there is no way to install this from a trusted source like MSDN.

Reply:
Also `%HOMEPATH%\.android\debug.keystore` is not found.

Reply:
I think we need to mention that MSAL will throw an exception that will contain the correct redirect URI so that people don't go crazy calculating this uri.

Reply:
`%LocalAppData%\Xamarin\Mono for Android\debug.keystore` is the correct path on Windows. Not sure for Mac.

Reply:
Unfortunately, this is the guidance from the azure portal on how to calculate the redirect URi on android. In our MSAL wiki, I have a much better explanation on how to get this value.